src/makeInfinityAdder.js

```diff
@@ -6,7 +6,38 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let currentSum = 0;
+  let count = 0;
+  const adder = (number) => {
+    if (!number) {
+      currentSum = 0;
+
+      return currentSum;
+    }
+
+    const result = (num) => {
+      if (num === undefined && count === 0) {
+        count++;
+
+        return currentSum;
+      }
+
+      if (num === undefined && count === 1) {
+        currentSum = 0;
+
+        return currentSum;
+      }
+      currentSum += num;
+
+      return result;
+    };
+
+    result(number);
+
+    return result;
```

Comment:
You don't need this large construction
```suggestion
    currentSum += number;
    
    return adder
```